### PR TITLE
HOTFIX repair broken setup for different cases

### PR DESCRIPTION
**Pull request by Boehrsi**
**Link to the given issue**
None

**Describe what the problem was / what the new feature is**
- Missing push registration during start
- Notification settings not loading
- Exceptions during entering a chat
- Wrongly started background service

**Describe your solution**

1. Start app
2. COI is supported: Set config values for COI (on first start)
3. Onboarding done or disabled
4. COI is supported: Register push (on first start) and start push, COI is not supported: Start background handling

**Additional context**
- Config now handles COI variables (previously located in the Customer area)
- Config hardened and cleaned up
- Removed the BuildContext requirement from the MainBloc
- Customer logic reduced to a minimum

### CODE REVIEW

File: lib/src/push/push_manager.dart [hunk 1]
```diff
@@ -68,9 +67,8 @@ class PushManager {
 
   PushManager._internal();
 
-  void setup(BuildContext buildContext) async {
-    this._buildContext = buildContext;
-    _pushBloc = BlocProvider.of<PushBloc>(_buildContext);
+  Future<void> setup(PushBloc pushBloc) async {
```
Comment (florianhaar) [inline lib/src/push/push_manager.dart:70]:
```suggestion
  Future<void> setupAsync(PushBloc pushBloc) async {
```
it is an async function so the naming should be "...Async"

File: lib/src/extensions/numbers_apis.dart [hunk 2]
```diff
@@ -21,6 +21,16 @@ extension Convert on int {
     }
     return "${result.toStringAsFixed(2)} $unit";
   }
+
+  bool toBool() {
```
[the remark this answers is not in the document]
Comment (Boehrsi) [reply lib/src/extensions/numbers_apis.dart:25]: Yep we can 👍 but in this case we should stay with the given API. Meaning that `num` already provides the methods `toDouble()`, `toSigned()` etc.. For the general usage of method vs. getter we should have a talk again nevertheless. For extensions I would would for "keeping new stuff close the current implementations".

File: lib/src/customer/customer_delegate.dart [hunk 3]
```diff
@@ -98,21 +98,16 @@ class CustomerDelegate with DynamicScreenCustomerDelegate {
...
           if (await Permission.notification.request().isGranted) {
-            _mainBloc = BlocProvider.of<MainBloc>(context);
-            await _mainBloc.setupManagers(context);
-            await setPreference(preferenceNotificationsPull, true);
-            BackgroundRefreshManager().start();
-
-          } else {
-            await setPreference(preferenceNotificationsPull, false);
-            BackgroundRefreshManager().stop();
+            if (_config.coiSupported) {
```
[the remark this answers is not in the document]
Comment (Boehrsi) [reply lib/src/customer/customer_delegate.dart:101]: Absolutely true, the only reason I used the current names is to stay aligned with the DCC naming, but probably using something "more correct" is better.

File: lib/src/data/config_extension.dart [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * for more details.
+ */
+
+class ConfigExtension {
+  static const coiSupported = "coi_supported";
+  static const coiEnabled = "coi_enabled";
+  static const coiMessageFilterEnabled = "coi_message_filter_enabled";
+
+  static const getAll = [coiSupported, coiEnabled, coiMessageFilterEnabled];
+}
```
[the remark this answers is not in the document]
Comment (Boehrsi) [reply lib/src/data/config_extension.dart:49]: Yep one should use the given tooling :D

**Comment by Boehrsi** [comment]
Applied requested fixes.

**Pull request merged**